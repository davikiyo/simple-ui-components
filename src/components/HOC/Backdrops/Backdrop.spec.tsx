import { render, screen } from '@testing-library/react'

import withBackDrop, { BackdropProps } from './Backdrop'

const TestComponent = ({ show = false }: BackdropProps) => (show ? <div>TEST</div> : null)

let WrappedComponent = withBackDrop(TestComponent)

beforeEach(() => {
  WrappedComponent = withBackDrop(TestComponent)
})

describe('Backdrop HOC', () => {
  it('should not display a text by default', () => {
    render(<WrappedComponent />)
    expect(screen.queryByText('TEST')).toBeNull()
  })

  it('should display a text with a dark backdrop', () => {
    const { asFragment } = render(<WrappedComponent show />)
    expect(screen.getByText('TEST')).toBeInTheDocument()
    expect(screen.getByTestId('backdrop-dark')).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })

  it('should display a text with a light backdrop', () => {
    WrappedComponent = withBackDrop(TestComponent, { color: 'light' })
    const { asFragment } = render(<WrappedComponent show />)
    expect(screen.getByTestId('backdrop-light')).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })

  it('should fix the content', () => {
    WrappedComponent = withBackDrop(TestComponent, { fixContent: true })
    render(<WrappedComponent show />)
    expect(document.body.style.overflow).toBe('hidden')
  })
})
