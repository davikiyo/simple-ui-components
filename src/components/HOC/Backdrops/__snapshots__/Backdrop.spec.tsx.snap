// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backdrop HOC should display a text with a dark backdrop 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    class="c-iJuALO c-iJuALO-dsZDrf-color-dark c-iJuALO-fTgyiV-show-true"
    data-testid="backdrop-dark"
  />
  <div
    data-testid="focus-trap"
    style="outline: none;"
    tabindex="0"
  >
    <div>
      TEST
    </div>
  </div>
</DocumentFragment>
`;

exports[`Backdrop HOC should display a text with a light backdrop 1`] = `
<DocumentFragment>
  <div
    aria-hidden="false"
    class="c-iJuALO c-iJuALO-kQAVZF-color-light c-iJuALO-fTgyiV-show-true"
    data-testid="backdrop-light"
  />
  <div
    data-testid="focus-trap"
    style="outline: none;"
    tabindex="0"
  >
    <div>
      TEST
    </div>
  </div>
</DocumentFragment>
`;
